keps/sig-apps/sidecarcontainers.md

```diff
@@ -162,11 +193,29 @@ Init containers would be able to have `sidecar: true` applied to them as it's an
 
 Older Kubelets that don't implement the sidecar logic could have a pod scheduled on them that has the sidecar field. As this field is just an addition to the Container Spec the Kubelet would still be able to schedule the pod, treating the sidecars as if they were just a normal container. This could potentially cause confusion to a user as their pod would not behave in the way they expect, but would avoid pods being unable to schedule.
 
+## Design Details
 
-## Graduation Criteria
-
+### Test Plan
 //TODO
 
+### Graduation Criteria
+#### Alpha -> Beta Graduation
+* Addressed feedback from Alpha testers
+* Thorough E2E and Unit testing in place
+
+#### Beta -> GA Graduation
+* Sufficient number of end users are using the feature
+* No significant bugs found for 2 beta releases
+
+### Upgrade / Downgrade Strategy
+When upgrading no changes should be needed to maintain existing behaviour as all of this behaviour is optional and disabled by default.
+To activate the feature they will need to enable the feature gate and mark their containers as sidecars in the container spec.
+
+When downgrading users will need to remove the sidecar field from their container specs as the Apiserver will not recognise the field.
```

Comment:
If I recall correctly, I think the apiserver will silently drop unknown fields. On the other hand, by default kubectl will try to block sending of unknown fields on the client side.

Reply:
Yeah I was confusing this with Kubectl, the api will just drop it. I'll rephrase this to be about kubectl.